### PR TITLE
bug/ handle RPCs that don't handle event filters properly

### PR DESCRIPTION
**Pull request by gsteenkamp89**
closes ACX-3028

- Attempts to first use eth_newFilter & eth_getFilterChanges
- If there is an error, revert to using getLogs & scraper

### CODE REVIEW

File: packages/sdk/src/utils/fetch.ts [hunk 1]
```diff
@@ -128,11 +128,8 @@ export const fetchAcrossApi = makeFetcher("Across API", (res, data, url) => {
...
 
-export const fetchIndexerApi = makeFetcher(
-  "Indexer API",
-  async (res, data, url) => {
-    if (typeof data === "object" && data !== null && "error" in data) {
-      throw new IndexerError(url, data?.message, data?.error);
-    }
-  },
-);
+export const fetchIndexerApi = makeFetcher("Indexer API", (res, data, url) => {
```
Comment (gsteenkamp89) [inline packages/sdk/src/utils/fetch.ts:131]: @dohaki this `async` was causing an unexpected context for catching exceptions

Comment (dohaki) [reply packages/sdk/src/utils/fetch.ts:131]: ahhh damn my bad 😓

File: packages/sdk/src/actions/waitForFillTx.ts [hunk 2]
```diff
@@ -14,11 +14,42 @@ export type WaitForFillTxParams = {
...
+    return statusFromFilter;
+  } catch (e) {
+    const message =
+      "Event filtering currently disabled for this RPC provider, switching to getFIllByDepositTx()";
+
+    params?.logger
+      ? params.logger.error(message, {
+          cause: e,
+        })
+      : console.error(message, {
```
Comment (dohaki) [inline packages/sdk/src/actions/waitForFillTx.ts:35]: nit: should we instead pass in the default console logger?

**Review by dohaki — APPROVED** [review]
Looks good! Left nits only

File: packages/sdk/src/actions/waitForFillTx.ts (hunk 2 above)
Comment (gsteenkamp89) [reply packages/sdk/src/actions/waitForFillTx.ts:35]: If this is being called via the client instance then the logger will be either a custom logger or the default logger and if not then we can just use the console.

**Pull request merged**